src/tests/ftest/performance/performance_test_base.py

```diff
@@ -0,0 +1,407 @@
+"""
+import os
+import time
+import types
+
+from ior_test_base import IorTestBase
+from mdtest_test_base import MdtestBase
+#from mdtest_utils import MdtestMetrics
+from general_utils import get_subprocess_stdout
+from ior_utils import IorMetrics
+from command_utils_base import EnvironmentVariables
+import oclass_utils
+
+# TODO dmg system query as non-json to reduce log clutter # pylint: disable=fixme
+
+class PerformanceTestBase(IorTestBase, MdtestBase):
+    # pylint: disable=too-many-ancestors
+    """Base performance class.
+
+    Optional yaml config values:
+        performance/phase_barrier_s (int): seconds to wait between IOR write/read phases.
+        performance/env (list): list of env vars to set for IOR/MDTest.
+
+    Outputs:
+        */data/performance.log: Contains input parameters and output metrics.
+        */data/daos_metrics/<host>_engine<idx>.csv: daos_metrics output for each host/engine
+    """
+
+    def __init__(self, *args, **kwargs):
+        """Initialize a PerformanceTestBase object."""
+        super().__init__(*args, **kwargs)
+
+        # Need to restart so daos_metrics and system state is fresh
+        self.start_servers_once = False
+
+        self.dmg_cmd = None
+        self.daos_cmd = None
+        self._performance_log_dir = self.outputdir
+        self._performance_log_name = os.path.join(self._performance_log_dir, "performance.log")
+        self.engines_per_host = 0
+        self.num_servers = 0
+        self.num_targets = 0
+        self.num_clients = 0
+        self.phase_barrier_s = 0
+        self.performance_env = EnvironmentVariables()
+
+    def setUp(self):
+        """Set up each test case."""
+        # Start the servers and agents
+        super().setUp()
+
+        self.dmg_cmd = self.get_dmg_command()
+        self.daos_cmd = self.get_daos_command()
+        self.engines_per_host = self.server_managers[0].get_config_value("engines_per_host") or 1
+        self.num_servers = self.engines_per_host * len(self.hostlist_servers)
+        self.num_targets = int(self.server_managers[0].get_config_value("targets"))
+        self.num_clients = len(self.hostlist_clients)
+        self.phase_barrier_s = self.params.get("phase_barrier_s", '/run/performance/*', 0)
+        self.__parse_performance_env()
+        self.__override_cmd_env(self.ior_cmd)
+        self.__override_cmd_env(self.mdtest_cmd)
+
+    def __parse_performance_env(self):
+        """Parse the performance environment variables.
+
+        This will be handled more cleanly by DAOS-9221. DO NOT COPY THIS.
+
+        For example:
+        performance:
+            env:
+                - D_LOG_MASK=ERR
+
+        """
+        perf_env = self.params.get("env", "/run/performance/*", {})
+        if not perf_env:
+            return
+        for key_val in perf_env:
+            key, val = key_val.split('=')
+            self.performance_env[key] = val
+
+    def __override_cmd_env(self, cmd):
```

Comment:
I want to stress that this is a hack until DAOS-9221 (blocked) introduces proper support